src/components/CommentList.jsx

```diff
@@ -8,33 +8,42 @@ import { useParams } from "react-router-dom";
 import Comment from "./Comment";
 import Modal from "./Modal";
 import SuccessMessage from "./SuccessMessage";
+import Loading from "./Loading";
 
 const CommentList = ({ username }) => {
   const [comments, setComments] = useState([]);
   const [commentText, setCommentText] = useState("");
+  const [isLoading, setIsLoading] = useState(true);
+  const [errorComment, setErrorComment] = useState(false);
   const [disable, setDisable] = useState(false);
   const { article_id } = useParams();
-
-  $(".message .close").on("click", function () {
-    $(this).closest(".message").transition("fade");
-  });
-
+  
   const handleSubmit = (event) => {
     event.preventDefault();
     const comment = {
       username: username,
       body: commentText,
     };
-    postCommentByArticleId(article_id, comment).then((data) => {
-      setComments((currComments) => {
-        return [data, ...currComments];
+    postCommentByArticleId(article_id, comment)
+      .then((data) => {
+        setComments((currComments) => {
+          return [data, ...currComments];
+        });
+      })
+      .catch((err) => {
```

Comment:
Convention should be put our catch block at the end of our promise chain 👍

Reply:
Rich thank you for the feedback, about the catch, if I place it at the end it's hard to check if there was an error or not and then I will be shown a successful message sometimes because the promise hasn't ended and I don't know if there was an error or not. If you know some way I could do this it would be great if you show me, please.

Reply:
Hmm so long as we have chained the promises correctly then the catch block should fire if placed at the end, no matter where in the promise chain the error has been thrown. Maybe one to play with at a later date but will likely just be a syntax issue somewhere 👍

Reply:
I will try like that now because before I was checking if there was an error outside the Promise so maybe that's why it was showing the successful message first.